src/runContract.js

```diff
@@ -131,31 +166,36 @@ export default async (event, context) => {
     })
 
     const transaction = new Transaction(xdr, Networks[process.env.STELLAR_NETWORK])
-    const hash = transaction.hash().toString('hex')
-
-    if (!find(transaction._operations, {
-      type: 'payment',
-      destination: process.env.TURRET_ADDRESS,
-      amount: new BigNumber(process.env.TURRET_RUN_FEE).toFixed(7),
-      asset: Asset.native()
-    })) throw 'Missing or invalid fee payment'
-
-    await server
-    .transactions()
-    .transaction(hash)
-    .call()
-    .catch((err) => err)
-    .then((err) => {
-      if (
-        err.response
-        && err.response.status === 404
-      ) return
-
-      else if (err.response)
-        throw err
-
-      throw 'Transaction has already been submitted'
-    })
+
+    if (
+      !event.headers['X-Payment']
+      && event.headers['X-Turrets']
+    ) {
+      const hash = transaction.hash().toString('hex')
+
+      if (!find(transaction._operations, {
+        type: 'payment',
+        destination: process.env.TURRET_ADDRESS,
+        amount: new BigNumber(process.env.TURRET_RUN_FEE).toFixed(7),
+        asset: Asset.native()
+      })) throw 'Missing or invalid fee payment'
+
+      await server
+      .transactions()
+      .transaction(hash)
+      .call()
+      .catch((err) => {
+        if (
+          err.response
+          && err.response.status === 404
+        ) return
+
+        else if (err.response)
+          throw err
+
+        throw 'Transaction has already been submitted'
+      })
+    }
```

Comment:
Only require a fee payment in the returned xdr if we're not using the payment method.